### ISSUE TITLE
UI tests for existing LeanPlum events

### ISSUE THREAD

**Issue by jhugman** [mozilla-mobile/fenix#11210]
Current manual tests on LeanPlum integration would suggest some events aren't being fired in to the LeanPlum SDK. 

It is not always clear that this is a missing work in fenix, bugs in fenix, bugs in the LeanPlum SDK or configuration errors in dashboarding with the marketing team.

In order to ameliorate communication between the mobile engagement team and engineering, such a test would be the final stage of sign-off/confirmation of a new event.

This issue is to come up with tests that can

* document and test the events that are currently supported.
* be obviously extendable when new events are added.
* be impossible to ignore when new events are added.

An end-to-end UI test would be most convincing, but proving just the fenix integration would be a useful place to start: 

* allow for a `SpyMetricController` to assert that the metrics controller is being called to log the event in existing UI tests.
* ensure that the `Events.name` declared in the `LeanPlumMetricService` logs the correct LeanPlum event name, as agreed with Mobile Engagement team.

Tasks:

* [ ] UI tests showing linkage between the UI events and the event reaching the lean plum SDK
* [ ] those tests being run on CI with the same cadence of other UI tests.
* [ ] a markdown file in the repo specifying when the event is fired and the name of the event.

**Comment by liuche** [comment]
@jhugman could you summarize what you've tried here? for whoever might need to pick this up.
